src/Microsoft.AspNet.Hosting/HostingEngine.cs

```diff
@@ -62,7 +66,13 @@ private void EnsureBuilder(HostingContext context)
                 return;
             }
 
+            EnsureApplicationServices(context);
+            if (_builderFactory == null)
```

Comment:
No, we need to ensure context.ApplicationServices is created here (don't really need it, since its the first thing we ensure in the Start now())